common/src/main/java/io/druid/audit/AuditManager.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Druid - a distributed column store.
+ * Copyright 2012 - 2015 Metamarkets Group Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.druid.audit;
+
+public interface AuditManager
+{
+  public void createAuditEntry(AuditEntry auditEntry);
```

Comment:
This interface makes it impossible to do audit logging in a transaction with the actual change. So there's a possibility of making a change without having an audit log entry for it, or of having an audit log entry made but no change in the actual table.

There's also inconsistency with how it's used: sometimes createAuditEntry is called before making the change in the main table, and sometimes after. I think if we're not going to do a transaction, it's best to always do the audit log first and then do the actual change. This is because I think it's better to have a phantom audit log entry than to have an audit-less change in the actual table.